include/deal.II/fe/fe_tools.h

```diff
@@ -841,148 +841,234 @@ namespace FETools
-                              const FiniteElement<dim,spacedim> *fe4=NULL,
-                              const unsigned int        N4=0,
-                              const FiniteElement<dim,spacedim> *fe5=NULL,
-                              const unsigned int        N5=0);
+    /**
+     * Compute the nonzero components for each shape function of a
+     * composed finite element described by a list of finite elements
+     * with multiplicities given in the second argument.
+     *
+     * If @p do_tensor_product is true, the number of components (and
+     * thus the size of the ComponentMask objects) is the sum over the
+     * product of the number of components in each of the finite
+     * elements times the corresponding multiplicity.  Otherwise the
+     * number of components is taken from the first finite element with
+     * non-zero multiplicity, and all other elements with non-zero
+     * multiplicities need to have the same number of vector components.
+     *
+     * See the documentation of namespace FETools::Compositing for more
+     * information about the @p do_tensor_product argument.
+     */
+    template <int dim, int spacedim>
+    std::vector<ComponentMask>
+    compute_nonzero_components (const std::vector<const FiniteElement<dim,spacedim>*> &fes,
+                                const std::vector<unsigned int>              &multiplicities,
+                                const bool do_tensor_product = true);
 
-  /**
-   * For a given (composite) @p finite_element build @p
-   * system_to_component_table, @p system_to_base_table and @p
-   * component_to_base_table.
-   *
-   * If @p do_tensor_product is true, the number of components
-   * used for the composite element is the sum over the
-   * product of the number of components in each of the finite
-   * elements times the corresponding multiplicity.  Otherwise the
-   * number of components is taken from the first finite element with
-   * non-zero multiplicity, and all other elements with non-zero
-   * multiplicities need to have the same number of vector components.
-   */
-  template <int dim, int spacedim>
-  void
-  build_cell_tables(std::vector< std::pair< std::pair< unsigned int, unsigned int >, unsigned int > > &system_to_base_table,
-                    std::vector< std::pair< unsigned int, unsigned int > >  &system_to_component_table,
-                    std::vector< std::pair< std::pair< unsigned int, unsigned int >, unsigned int > > &component_to_base_table,
-                    const FiniteElement<dim,spacedim> &finite_element,
-                    const bool do_tensor_product = true);
+    /**
+     * Compute the non-zero vector components of a composed finite
+     * element. This function is similar to the previous one, except
+     * that the pointers indicate the elements to be composed, and the
+     * arguments @p N1, @p N2, ... the multiplicities. Null pointers
+     * indicate that an argument is to be skipped.
```

Comment:
Aw, this is a bit awkward here since the other version of this function *does* have the `do_tensor_product` argument. I'll add the argument here as well.